dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/imports/BodyConverter.java

```diff
@@ -121,7 +121,7 @@ public TrackerBundleParams convert( TrackerBundleParams dataBundle )
 
         // Set UID for all enrollments and notes
         dataBundle.getEnrollments().stream()
-            .peek( enrollment -> updateEnrollmentReferences( enrollment, enrollment.getTrackedEntity() ) )
```

Comment:
All update methods were working with side effects and returning void so peek was the only choice.
Changing to map as peek is a code smell and it can behave in unpredictable ways